heimdall-gateway/src/main/java/br/com/conductor/heimdall/gateway/filter/helper/HttpImpl.java

```diff
@@ -1,261 +1,305 @@
+	}
+
+	@Override
+	public ApiResponseImpl sendPut() {
+
+		ResponseEntity<String> entity;
+		if (headers.isEmpty()) {
+
+			requestBody = new HttpEntity<>(body);
+			entity = rest().exchange(uriComponentsBuilder.build().encode().toUri(), HttpMethod.PUT, requestBody,
+					String.class);
+		} else {
+
+			if (Objeto.notBlank(formData)) {
+
+				HttpEntity<MultiValueMap<String, String>> request = new HttpEntity<>(formData, headers);
+				entity = rest().exchange(uriComponentsBuilder.build().encode().toUri(), HttpMethod.PUT, request,
+						String.class);
+			} else {
+
+				requestBody = new HttpEntity<>(body, headers);
+				entity = rest().exchange(uriComponentsBuilder.build().encode().toUri(), HttpMethod.PUT, requestBody,
+						String.class);
+			}
+		}
+		ApiResponseImpl apiResponse = new ApiResponseImpl();
+		apiResponse.setHeaders(entity.getHeaders().toSingleValueMap());
+		apiResponse.setBody(entity.getBody());
+		apiResponse.setStatus(entity.getStatusCodeValue());
+
+		return apiResponse;
+	}
+
+	@Override
+	public ApiResponseImpl sendDelete() {
+
+		ResponseEntity<String> entity;
+
+		if (headers.isEmpty()) {
+			entity = rest().exchange(uriComponentsBuilder.build().encode().toUri(), HttpMethod.DELETE, null,
+					String.class);
+		} else {
+			entity = rest().exchange(uriComponentsBuilder.build().encode().toUri(), HttpMethod.DELETE,
+					new HttpEntity<>(headers), String.class);
+		}
+
+		ApiResponseImpl apiResponse = new ApiResponseImpl();
+		apiResponse.setHeaders(entity.getHeaders().toSingleValueMap());
+		apiResponse.setBody(entity.getBody());
+		apiResponse.setStatus(entity.getStatusCodeValue());
+
+		return apiResponse;
+	}
+
+	@Override
+	public ApiResponseImpl sendPatch() {
+
+		ResponseEntity<String> entity;
+
+		if (headers.isEmpty()) {
+			requestBody = new HttpEntity<>(body);
+			entity = rest().exchange(uriComponentsBuilder.build().encode().toUri(), HttpMethod.PATCH, requestBody,
+					String.class);
+		} else {
+			if (Objeto.notBlank(formData)) {
+				HttpEntity<MultiValueMap<String, String>> request = new HttpEntity<>(formData, headers);
+				entity = rest().exchange(uriComponentsBuilder.build().encode().toUri(), HttpMethod.PATCH, request,
+						String.class);
+			} else {
+				requestBody = new HttpEntity<>(body, headers);
+				entity = rest().exchange(uriComponentsBuilder.build().encode().toUri(), HttpMethod.PATCH, requestBody,
+						String.class);
+			}
+		}
+
+		ApiResponseImpl apiResponse = new ApiResponseImpl();
+		apiResponse.setHeaders(entity.getHeaders().toSingleValueMap());
+		apiResponse.setBody(entity.getBody());
+		apiResponse.setStatus(entity.getStatusCodeValue());
+
+		return apiResponse;
+	}
+
+	@Override
+	public RestTemplate clientProvider(RestTemplate restTemplate) {
+
+		this.restTemplate = restTemplate;
+		return this.restTemplate;
+	}
+
+	private RestTemplate rest() {
+		if (this.restTemplate == null) {
+			// Create a RestTemplate from an HttpComponentsClientHttpRequestFactory. This is

```

Comment:
@zeroberto you don't need to explain the factory implementation, remove comments plz.